docs/register-bitcoin-stake.md

```diff
@@ -0,0 +1,698 @@
+  bytes slashing_tx = 9 [ (gogoproto.customtype) = "BTCSlashingTx" ];
+  // delegator_slashing_sig is the signature on the slashing tx by the delegator (i.e., SK corresponding to btc_pk).
+  // It will be a part of the witness for the staking tx output.
+  // The staking tx output further needs signatures from covenant and finality provider in
+  // order to be spendable.
+  bytes delegator_slashing_sig = 10 [ (gogoproto.customtype) = "github.com/babylonlabs-io/babylon/types.BIP340Signature" ];
+  // unbonding_time is the time lock used when funds are being unbonded. It is be used in:
+  // - unbonding transaction, time lock spending path
+  // - staking slashing transaction, change output
+  // - unbonding slashing transaction, change output
+  // It must be smaller than math.MaxUInt16 and larger that max(MinUnbondingTime, CheckpointFinalizationTimeout)
+  uint32 unbonding_time = 11;
+  // fields related to unbonding transaction
+  // unbonding_tx is a bitcoin unbonding transaction i.e transaction that spends
+  // staking output and sends it to the unbonding output
+  bytes unbonding_tx = 12;
+  // unbonding_value is amount of satoshis locked in unbonding output.
+  // NOTE: staking_value and unbonding_value could be different because of the difference between the fee for staking tx and that for unbonding
+  int64 unbonding_value = 13;
+  // unbonding_slashing_tx is the slashing tx which slash unbonding contract
+  // Note that the tx itself does not contain signatures, which are off-chain.
+  bytes unbonding_slashing_tx = 14 [ (gogoproto.customtype) = "BTCSlashingTx" ];
+  // delegator_unbonding_slashing_sig is the signature on the slashing tx by the delegator (i.e., SK corresponding to btc_pk).
+  bytes delegator_unbonding_slashing_sig = 15 [ (gogoproto.customtype) = "github.com/babylonlabs-io/babylon/types.BIP340Signature" ];
+}
+```
+#### Explanation of Fields
+
+* `staker_addr`:
+  A Bech32-encoded Babylon address (`bbn...`) representing the
+  staker's Babylon account where staking rewards will be accumulated.
+  *This should be the same address that signs the registration transaction*.
+* `pop` (Proof of Possession):
+  A cryptographic signature proving that the submitter of the registration
+  transaction is the owner of the Bitcoin private key used for staking.
+  * `btc_sig_type`: Specifies the signature algorithm used.
+    The options are:
+    * `0` for [BIP-340 (Schnorr Signatures)](https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki)
+    * `1` for [BIP-322 (Generic Signing format)](https://github.com/bitcoin/bips/blob/master/bip-0322.mediawiki)
+      * Note that the [simple](https://github.com/bitcoin/bips/blob/master/bip-0322.mediawiki#simple) signature format is used.
+    * `2` for [ECDSA (Elliptic Curve Digital Signature Algorithm)](https://github.com/bitcoin/bips/blob/master/bip-0137.mediawiki)
+  * `btc_sig`: The signature generated by signing the `SHA-256` hash of the Babylon staker
```

Comment:
this is not exactly true. 

Currently `BIP340` and `BIP322` verify signature over the hash of staker address bytes, but ECDSA verify signature over `hex` encoded string of staker address bytes